std/format.d

```diff
@@ -3631,30 +3645,30 @@ if (hasToString!(T, Char))
 {
     enum overload = hasToString!(T, Char);
 
-    static if (overload == 6)
+    static if (overload == HasToStringResult.customPutWriterFormatSpec)
     {
         val.toString(w, f);
     }
-    else static if (overload == 5)
+    else static if (overload == HasToStringResult.customPutWriter)
     {
         val.toString(w);
     }
     // not using the overload enum to not break badly defined toString overloads
     // e.g. defining the FormatSpec as ref and not const ref led this function
     // to ignore that toString overload
-    else static if (is(typeof(val.toString((const(char)[] s){}, f))))
+    else static if (overload == HasToStringResult.constCharSinkFormatSpec)
```

Comment:
This is a breaking change, see comment above

Reply:
Why would this be a breaking change?